pkg/machine/machineset.go

```diff
@@ -0,0 +1,581 @@
+package machine
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// SetBuilder provides a struct for MachineSet object from the cluster and a MachineSet definition.
+type SetBuilder struct {
+	// SetBuilder definition. Used to create
+	// MachineSet object with minimum set of required elements.
+	Definition *machinev1beta1.MachineSet
+	// Created SetBuilder object on the cluster.
+	Object *machinev1beta1.MachineSet
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+	// errorMsg is processed before SetBuilder object is created.
+	errorMsg string
+	// string to store the public cloud
+	publicCloud string
+}
+
+const (
+	AwsCloud   = "aws"
+	GcpCloud   = "gcp"
+	AzureCloud = "azure"
+)
+
+// NewSetBuilderFromCopy returns an SetBuilder struct from a copied MachineSet.
+func NewSetBuilderFromCopy(
+	apiClient *clients.Settings,
+	nsName string,
+	instanceType string,
+	replicas int32,
+	workerLabel string) *SetBuilder {
```

Comment:
Please keep argument of the same type close to each other.

Reply:
Done